soda/core/soda/execution/anomaly_metric_check.py

```diff
@@ -48,53 +56,68 @@ def __init__(
 
     def evaluate(self, metrics: dict[str, Metric], historic_values: dict[str, object]):
         if self.skip_anomaly_check:
-            self.logs.error(
+            error_message = (
                 "Anomaly detection was not given a threshold. You might want to check if the parser returned errors"
             )
+            self.logs.error(error_message)
+            self.outcome = None
+            self.force_send_results_to_cloud = True
+            self.outcome_reasons = CheckOutcomeReasons(
+                queryFailed=ParserFailed(message=error_message, severity="error")
+            )
+            return
+
+        # TODO Review the data structure and see if we still need the KEY_HISTORIC_*
+        historic_measurements = historic_values.get(KEY_HISTORIC_MEASUREMENTS, {}).get("measurements", {})
+        historic_check_results = historic_values.get(KEY_HISTORIC_CHECK_RESULTS, {}).get("check_results", {})
+
+        if not historic_measurements:
+            warning_message = "Skipping anomaly metric check eval because there is no historic data yet!"
+            self.logs.warning(warning_message)
             self.outcome = None
-        else:
-            # TODO Review the data structure and see if we still need the KEY_HISTORIC_*
-            historic_measurements = historic_values.get(KEY_HISTORIC_MEASUREMENTS).get("measurements")
-            historic_check_results = historic_values.get(KEY_HISTORIC_CHECK_RESULTS).get("check_results")
-            if historic_measurements is None:
-                self.logs.warning("Skipping anomaly metric check eval because there is not enough historic data yet")
-                return
-
-            # Append current results
-            historic_measurements.get("results").append(
-                {
-                    "id": 61,  # Placeholder number that will be overwritten
-                    "identity": metrics[self.name].identity,
-                    "value": self.get_metric_value(),
-                    "dataTime": (
-                        self.data_source_scan.scan._data_timestamp.replace(tzinfo=timezone.utc).strftime(
-                            "%Y-%m-%dT%H:%M:%SZ"
-                        )
-                    ),
-                }
+            self.force_send_results_to_cloud = True
+            self.outcome_reasons = CheckOutcomeReasons(
+                notEnoughHistory=NotEnoughHistory(message=warning_message, severity="warn")
```

Comment:
since I was serializing dataclass to dict, I had to use notEnoughHistory since it was expected from the backend. However, with Tom's suggestion, I won't need to serialize it anymore.